src/debugpy/common/messaging.py

```diff
@@ -1211,11 +1211,17 @@ def wait(self):
         message handlers to finish executing.
         """
         parser_thread = self._parser_thread
-        if parser_thread is not None:
-            parser_thread.join()
-        handler_thread = self._handler_thread
-        if handler_thread is not None:
-            handler_thread.join()
+        try:
+            if parser_thread is not None:
+                parser_thread.join()
+        except AssertionError:
+            log.debug("Handled error joining parser thread.")
```

Comment:
This is handling the traceback:
```
Traceback (most recent call last):
  File "C:\Programs\pypy\lib-python\3\runpy.py", line 196, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Programs\pypy\lib-python\3\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\Users\username\.vscode-insiders\extensions\ms-python.python-2021.5.797839398-dev\pythonFiles\lib\python\debugpy\launcher\__main__.py", line 95, in <module>le>
    main()
  File "c:\Users\username\.vscode-insiders\extensions\ms-python.python-2021.5.797839398-dev\pythonFiles\lib\python\debugpy\launcher\__main__.py", line 52, in main
    launcher.channel.wait()
  File "c:\Users\username\.vscode-insiders\extensions\ms-python.python-2021.5.797839398-dev\pythonFiles\lib\python\debugpy\launcher/../..\debugpy\common\messaging.py"py", line 1215, in wait
    parser_thread.join()
  File "C:\Programs\pypy\lib-python\3\threading.py", line 1044, in join
    self._wait_for_tstate_lock()
  File "C:\Programs\pypy\lib-python\3\threading.py", line 1062, in _wait_for_tstate_lock
    self._stop()
  File "C:\Programs\pypy\lib-python\3\threading.py", line 996, in _stop
    assert not lock.locked()
AssertionError
```

It doesn't really seem `debugpy` is doing anything wrong and I don't really understand why sometimes the `join` fails under `PyPy` (it seems like an issue on `PyPy` for me, not on `debugpy`), so, just silencing the error...